packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -731,12 +746,17 @@ export function arraySchema<
 			setFlexNode(proxy, flexNode);
 			return proxy as unknown as schema;
 		}
+
+		public toJSON(): unknown {
+			// This override causes the class instance to `JSON.stringify` as `[a, b]` rather than `{0: a, 1: b}`.
+			return Array.from(this as unknown as TreeArrayNode);
```

Comment:
This is progress, so its fine for now (its nice to seperate fixes into their own change), but eventually we really should move the other methods into here so these casts can go away along with the defineProperties call below similar to how https://github.com/microsoft/FluidFramework/pull/20674 did it for maps nodes.